src/graphnet/data/curated_datamodule.py

```diff
@@ -271,12 +280,282 @@ def prepare_data(self) -> None:
+        # Static Member Variables:
+        self._mode = mode
+        if self._mode == "train":
+            self._pulsemaps = ["merged_photons"]
+        elif self._mode == "test":
+            self._pulsemaps = ["pulses"]
+        elif self._mode == "test-no-noise":
+            self._pulsemaps = ["pulses_no_noise"]
+        else:
+            raise AssertionError(
+                "'mode' must be one of "
+                f"{{['train', 'test', 'test-no-noise']}}"
+                f"got '{mode}'"
+            )
+        self._truth_table = "mc_truth"
+        self._event_truth = [
+            "interaction",
+            "initial_state_energy",
+            "initial_state_type",
+            "initial_state_zenith",
+            "initial_state_azimuth",
+            "initial_state_x",
+            "initial_state_y",
+            "initial_state_z",
+            "visible_inelasticity",
+        ]
+        self._pulse_truth = "pulses"
+        self._features = [
+            "sensor_pos_x",
+            "sensor_pos_y",
+            "sensor_pos_z",
+            "t",
+            "charge",
+            "string_id",
+        ]
+
+        ERDAHostedDataset.__init__(
+            self,
+            graph_definition=graph_definition,
+            download_dir=download_dir,
+            backend=backend,
+            train_dataloader_kwargs=train_dataloader_kwargs,
+            validation_dataloader_kwargs=validation_dataloader_kwargs,
+            test_dataloader_kwargs=test_dataloader_kwargs,
+        )
+
+    def _prepare_args(
+        self, backend: str, features: List[str], truth: List[str]
+    ) -> Tuple[Dict[str, Any], Union[List[int], None], Union[List[int], None]]:
+        """Prepare arguments for dataset.
+
+        Args:
+            backend: backend of dataset. Either "parquet" or "sqlite".
+            features: List of features from user to use as input.
+            truth: List of event-level truth variables from user.
+
+        Returns: Dataset arguments, train/val selection, test selection
+        """
+        if backend == "sqlite":
+            dataset_path = glob(os.path.join(self.dataset_dir, "*.db"))
+            # Cast from list to string if just 1 path
+            if isinstance(dataset_path, list) & len(dataset_path) == 1:
+                dataset_path: str = dataset_path[0]  # type: ignore
+
+            if self._mode == "train":
+                train_val = pd.read_parquet(
+                    os.path.join(
+                        self.dataset_dir,
+                        "selections",
+                        "train_selection.parquet",
+                    )
+                )["event_no"].tolist()
+                test = None
+            elif self._mode == "test":
+                train_val = None
+                test = pd.read_parquet(
+                    os.path.join(
+                        self.dataset_dir,
+                        "selections",
+                        "test_noise_selection.parquet",
```

Comment:
Calling this one "test_noise" instead of calling the no noise mode "test-no-noise" after having specified "no-noise" throughout the code is a bit confusing